src/Client.ts

```diff
@@ -0,0 +1,118 @@
+// tslint:disable-next-line import-name no-require-imports
+import { all } from 'deepmerge';
+import * as querystring from 'querystring';
+
+import { Provider } from './providers/Provider';
+import {
+    IOptionalUrlRequestOptions,
+    IRequestOptions,
+    IRequestRunner,
+    IResponse,
+} from './RequestRunner';
+
+// DO NOT EDIT, THIS IS UPDATE BY THE BUILD SCRIPT
+const packageVersion = '0.13.0'; // package version
+
+/**
+ * Main client.
+ */
+export class Client {
+    private provider: Provider;
+    private userAgent: string;
+    public urls = {
+        api: 'https://beam.pro/api/v1',
+        public: 'https://beam.pro',
+    };
+    /**
+     * The primary Beam client, responsible for storing authentication state
+     * and dispatching requests to the API.
+     */
+    constructor(private requestRunner: IRequestRunner) {
```

Comment:
It's not much work, but it's cognitive load and fiddling for something that no developer will ever have to deal with or even care about. It's useful for unit testing internally, but consumers of the library should stub or mock the library, not the request runner